test/blackbox-tests/test-cases/pkg/ocamlformat/solving-fails.t

```diff
@@ -0,0 +1,42 @@
+When an OCamlFormat version does not exists, $ dune fmt would fail with a
+solving error.
+
+  $ . ../helpers.sh
+  $ mkrepo
+
+Update with no dependency on OCamlFormat
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package
+  >  (name foo))
+  > EOF
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.13)
+  > (lock_dir
+  >  (path "dev-tools.locks/ocamlformat")
+  >  (repositories mock))
+  > (lock_dir
+  >  (repositories mock))
+  > (repository
+  >  (name mock)
+  >  (source "file://$(pwd)/mock-opam-repository"))
+  > EOF
+
+Update ".ocamlformat" file with unknown version of OCamlFormat.
+  $ cat > .ocamlformat <<EOF
+  > version = 0.26.9
+  > EOF
+
+An important cleaning here
```

Comment:
A more meaningful comment would explain why it's important to clean here.

Reply:
Normally this clean is not necessary.

We need to fix this in the PR, because when the dev-tool is disabled, even if `dev-tools.locks/ocamlformat` exists `dune fmt` is not supposed wokrs as if dev-tool is enabled.

Reply:
Makes sense. Can you update the comment to explain this.

Reply:
We could just extends the tests for dev-tool and remove this cleaning. I could do that.

Reply:
For now let's just prioritize getting this PR merged, so I'd say just update the comment to clarify and make additional changes in a later PR.

Reply:
You're right.